tools/RAiDER/delayFcns.py

```diff
@@ -167,7 +155,64 @@ def make_interpolator(xs, ys, zs, data):
     )
 
 
-def process_chunk(k, chunkInds, SP, SLV, in_shape, stepSize, ifWet, ifHydro, max_len, wm_file):
+def chunk(chunkSize, in_shape):
+    '''
+    Create a set of indices to use as chunks
+    '''
+    startInds =  makeChunkStartInds(chunkSize, in_shape)
+    chunkInds = makeChunksFromInds(startInds, chunkSize, in_shape)
+    return chunkInds
+
+
+def makeChunksFromInds(startInd, chunkSize, in_shape):
+    '''
+    From a list of tuples containing starting indices and
+    a tuple of chunkSize (length equal to ndim), create a 
+    list of arrays to be used as chunk indices
+    '''
+    indices = []
+    for ci in startInd:
+        index = []
+        for si, k, dim in zip(ci, chunkSize, range(len(chunkSize))):
+            if si+k > in_shape[dim]:
+                dend = in_shape[dim]
+            else:
+                dend = si+k
+            index.append(np.array(range(si, dend)))
+        indices.append(index)
+
+    # Now create the index mesh (for Ndim > 1)
+    chunks = []
+    if len(in_shape) > 1:
+        for index in indices:
+            chunks.append([np.array(g) for g in zip(*list(itertools.product(*index)))])
+    else:
+        chunks = indices
+
+    return chunks 
+
+
+def makeChunkStartInds(chunkSize, in_shape):
+    '''
+    Create a list of indices for chunking a 2D array
```

Comment:
If it only works for 2d arrays then that should probably be reflected in the function name. Sometimes I also like to assert my preconditions at the start of a function to make sure no programming errors occur.

```python3
assert len(in_shape) == 2
```